### PR TITLE
Replace PGI 19.4 with 19.10

### PR DESCRIPTION
**Pull request by pramodk**

### CODE REVIEW

**Comment by pramodk** [comment]
Retest this please

**Review by matz-e — CHANGES_REQUESTED** [review]
Could you please add the `version` for 19.10 to the `pgi` package?

**Comment by pramodk** [comment]
oh yes! missed that! (long gap to work with Spack! :smiley: )

**Review by iomaganaris — DISMISSED** [review]
LGTM

**Review by matz-e — DISMISSED** [review]
Looks better!

**Review by matz-e — DISMISSED** [review]

File: bin/spack [hunk 1]
```diff
@@ -11,7 +11,7 @@
 # See https://stackoverflow.com/a/47886254
 """:"
 # prefer python3, then python, then python2
-for cmd in python3 python python2; do
+for cmd in python3.7 python python2; do
```
Comment (matz-e) [inline bin/spack]: Why this? I don't think I'm OK with that.

Comment (pramodk) [reply bin/spack]: That’s my local change got pushed, I am still debugging this! Will cleanup once ready.

**Comment by pramodk** [comment]
@matz-e : ready for review and merge.

File: deploy/deploy.lib [hunk 2]
```diff
@@ -312,9 +313,8 @@ configure_compilers() {
             #update pgi modules for network installation
             PGI_DIR=$(dirname $(which makelocalrc))
             makelocalrc ${PGI_DIR} -gcc ${GCC_DIR}/bin/gcc -gpp ${GCC_DIR}/bin/g++ -g77 ${GCC_DIR}/bin/gfortran -x -net
-
             #configure pgi network license
-            template=$(find $PGI_DIR -name localrc* | tail -n 1)
+            template=$(find $PGI_DIR -name localrc* ! -iname "*.bak" | tail -n 1)
```
Comment (matz-e) [inline deploy/deploy.lib:317]: That's nasty. Did this fix your bugs?

**Review by matz-e — APPROVED** [review]

File: deploy/deploy.lib (hunk 2 above)
Comment (pramodk) [reply deploy/deploy.lib:317]: yes! Seems like makelocalrc now save the backup of previous config which is with gcc4.8.

**Pull request merged**